src/t_stream.c

```diff
@@ -1493,6 +1626,11 @@ size_t streamReplyWithRange(client *c, stream *s, streamID *start, streamID *end
     while(streamIteratorGetID(&si,&id,&numfields)) {
         /* Update the group last_id if needed. */
         if (group && streamCompareID(&id,&group->last_id) > 0) {
+            if (group->offset && streamIsContiguousRange(s,&id)) {
+                group->offset++;
```

Comment:
maybe you should update the 1,2,3 list of additional tasks this method does in the top comment.

Reply:
Added 4 to 1,2,3.